test/mock/createAPIResponse.ts

```diff
@@ -1,21 +1,41 @@
 import { SearchForFacetValues, Response, MultiResponse } from 'algoliasearch';
 
-export const createSingleSearchResponse = (
-  args: Partial<Response> = {}
-): Response => ({
-  hits: [],
-  nbHits: 0,
-  page: 0,
-  nbPages: 0,
-  hitsPerPage: 4,
-  processingTimeMS: 1,
-  query: '',
-  params: '',
-  index: 'index_name',
-  ...args,
-});
+export function createSingleSearchResponse<THit = any>(
+  subset: Partial<Response<THit>> = {}
+): Response<THit> {
+  const {
+    page = 0,
+    hitsPerPage = 20,
+    nbHits,
+    nbPages,
+    processingTimeMS = 0,
+    hits = [],
+    query = '',
+    params = '',
+    exhaustiveNbHits = true,
+    exhaustiveFacetsCount = true,
+    ...rest
+  } = subset;
+  const fallbackNbHits = nbHits !== undefined ? nbHits : hits.length;
+  const fallbackNbPages =
+    nbPages !== undefined ? nbPages : Math.ceil(fallbackNbHits / hitsPerPage);
```

Comment:
I don't think the tests are useful for this. We could leverage the default value to make them easier to understand. 

```ts
export function createSingleSearchResponse() {
  const {
    hits = [],
    hitsPerPage = 20,
    nbHits = hits.length,
    nbPages = Math.ceil(nbHits / hitsPerPage)
  } = args;

  // ...
}
```

Reply:
had it there earlier, but thought it might have been too complicated. Seeing it written out again, I'll flip back to that form